packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2550,6 +2550,12 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
         super.paint,
         Offset.zero,
       );
+    } else if (selection!.isCollapsed) {
```

Comment:
@justinmc I'm wondering if this change is worth while, i'm not seeing any issues without it. Previously when the selection was collapsed `SelectionOverlay` would build an empty end handle and `RenderEditable` would only push a layer for the start handle. Now `SelectionOverlay` may build and empty end handle or start handle depending on certain conditions. 

When the selection is collapsed with no handles being dragged, then build an empty end handle, this case would be alright without this code since the end handle is not built.

When the selection is collapsed with the start handle being dragged, then build an empty end handle, this case would also be fine without this code since the end handle is not built.

When the selection is collapsed with the end handle being dragged, then build an empty start handle, this case might not be fine without this code since we are dragging the end handle and its layer won't be there.

Reply:
I'm leaning towards keeping this in for the case of:
> When the selection is collapsed with the end handle being dragged, then without the code only the start handle layer is pushed. This seems wrong since the end handle is being dragged and its layer will not be there.

`RenderParagraph` does something similar (pushes both layers when collapsed). https://github.com/flutter/flutter/blob/ae736e2cc325998df328595e039208d545128265/packages/flutter/lib/src/rendering/paragraph.dart#L3086C1-L3105C6

Reply:
Sounds good.